File: program63.java
///////////////////////////////////////////////////////
//
//  File name :     program63.java
//  Descreption :   Accept Input and Calculate Summation of Non-Factors of Number and Display
//  Author :        Vaibhav
//  Date :          
//
///////////////////////////////////////////////////////

import java.util.*;

class Number
{
    public int SumNonFactors(int iNo)
    {
        int iCnt = 0;
        int iSum = 0;
        for(iCnt = 1;iCnt <= iNo;iCnt++)
        {
            if((iNo % iCnt) != 0)       //dry run
            {
                iSum = iSum+ iCnt;
            }
        }
        return iSum;
    }
}

class program63
{
    public static void main(String A[])
    {
        Scanner sobj = new Scanner(System.in);

        System.out.println("---- This Application is used to Display Summation of Non-Factors of Number ----");

        System.out.println("Enter number: ");
        int iValue = sobj.nextInt();

        Number nobj = new Number();

        int iRet = nobj.SumNonFactors(iValue) ;

        System.out.println("Summation of Non factors are : "+iRet);
    }
}

File: Java/program63.c
///////////////////////////////////////////////////////
//
//  File name :     program63.c
//  Descreption :   Accept Input and Calculate Summation of Non-Factors of Number and Display
//  Author :        Vaibhav
//  Date :          20/05/2025
//
///////////////////////////////////////////////////////

#include<stdio.h>

int DisplaySumNonFactors(int iValue)
{
    int iCnt = 0;
    int iSum = 0;

    for(iCnt = 1;iCnt <= iValue; iCnt++)
    {
        if((iValue % iCnt) != 0)
        {
            iSum = iSum + iCnt;
        }
    }
    return iSum;
}

int main()
{
    int iNo = 0;
    int iRet = 0;

    printf("---- This Application is used to used to Display Summation of Non-Factors of Number ----\n");

    printf("Enter Number: ");
    scanf("%d",&iNo);

    iRet = DisplaySumNonFactors(iNo);

    printf("Summation of Non Factors are : %d",iRet);
    
    return 0;
}

